flocker/node/test/test_deploy.py

```diff
@@ -2362,13 +2434,16 @@ def test_app_with_changed_ports_restarted(self):
             current_cluster_state=EMPTY,
         )
 
-        expected = Sequentially(changes=[InParallel(changes=[
-            Sequentially(changes=[
-                StopApplication(application=old_postgres_app),
-                StartApplication(application=new_postgres_app,
-                                 hostname="node1.example.com")
+        expected = Sequentially(changes=[
+            OpenPorts(ports=frozenset([50433])),
+            InParallel(changes=[
+                Sequentially(changes=[
+                    StopApplication(application=old_postgres_app),
+                    StartApplication(application=new_postgres_app,
+                                     hostname="node1.example.com")
                 ]),
-        ])])
+            ]),
+        ])
 
         self.assertEqual(expected, result)
 
```

Comment:
You should write a unit test or two for the `OpenPorts` class, running against the in-memory network object.